claims_p1.go

```diff
@@ -215,99 +276,147 @@ func (c P1Claims) ToUnvalidatedJSON() ([]byte, error) { //nolint:gocritic
+	}
+
+	l := len(*c.BootSeed)
+	if l != 32 {
+		return nil, fmt.Errorf(
+			"%w: invalid length %d (MUST be 32 bytes)",
+			ErrWrongSyntax, l,
+		)
+	}
+
+	return *c.BootSeed, nil
 }
 
 func (c P1Claims) GetCertificationReference() (string, error) { //nolint:gocritic
-	return getCertificationReference(c.CertificationReference, PsaProfile1)
+	if c.CertificationReference == nil {
+		return "", ErrOptionalClaimMissing
+	}
+
+	if !CertificationReferenceP1RE.MatchString(*c.CertificationReference) &&
+		!CertificationReferenceP2RE.MatchString(*c.CertificationReference) {
+		return "", fmt.Errorf(
+			"%w: MUST be in EAN-13 or EAN-13+5 format",
+			ErrWrongSyntax,
+		)
+	}
+
+	return *c.CertificationReference, nil
 }
 
 // Caveat: this may return nil on success if psa-no-sw-measurement is asserted
-func (c P1Claims) GetSoftwareComponents() ([]SwComponent, error) { //nolint:gocritic
-	v := c.SwComponents
-	f := c.NoSwMeasurements
-
-	if v == nil {
+func (c P1Claims) GetSoftwareComponents() ([]ISwComponent, error) { //nolint:gocritic
+	if c.SwComponents == nil || c.SwComponents.IsEmpty() {
 		// psa-no-sw-measurement must be asserted
-		if f == nil {
-			return nil, ErrMandatoryClaimMissing
+		if c.NoSwMeasurements == nil {
+			return nil, fmt.Errorf("%w (MUST have at least one sw component or no-sw-measurements set)",
+				ErrMandatoryClaimMissing)
 		}
 		return nil, nil
 	}
 
 	// psa-no-sw-measurement must not be asserted at the same time as psa-software-components
-	if f != nil {
+	if c.NoSwMeasurements != nil {
 		return nil,
 			fmt.Errorf(
 				"%w: psa-no-sw-measurement and psa-software-components cannot be present at the same time",
-				ErrWrongClaimSyntax,
+				ErrWrongSyntax,
 			)
 	}
 
-	if err := isValidSwComponents(*v); err != nil {
-		return nil, err
-	}
-
-	return *v, nil
+	return c.SwComponents.Values()
 }
 
 func (c P1Claims) GetNonce() ([]byte, error) { //nolint:gocritic
-	v := c.Nonce
-
-	if v == nil {
+	if c.Nonce == nil {
 		return nil, ErrMandatoryClaimMissing
 	}
 
-	if err := isValidNonce(*v); err != nil {
+	if err := ValidateNonce(*c.Nonce); err != nil {
 		return nil, err
 	}
 
-	return *v, nil
+	return *c.Nonce, nil
 }
 
 func (c P1Claims) GetInstID() ([]byte, error) { //nolint:gocritic
-	v := c.InstID
-
-	if v == nil {
+	if c.InstID == nil {
 		return nil, ErrMandatoryClaimMissing
 	}
 
-	if err := isValidInstID(*v); err != nil {
+	if err := ValidateInstID(*c.InstID); err != nil {
 		return nil, err
 	}
 
-	return *v, nil
+	return *c.InstID, nil
 }
 
 func (c P1Claims) GetVSI() (string, error) { //nolint:gocritic
-	return getVSI(c.VSI)
-}
+	if c.VSI == nil {
+		return "", ErrOptionalClaimMissing
+	}
 
-func (c P1Claims) GetConfig() ([]byte, error) { //nolint:gocritic
+	if err := ValidateVSI(*c.VSI); err != nil {
+		return "", err
+	}
 
-	return nil, fmt.Errorf("invalid GetConfig invoked on p1 claims")
+	return *c.VSI, nil
 }
 
-func (c P1Claims) GetHashAlgID() (string, error) { //nolint:gocritic
+func init() {
```

Comment:
This appears to be a local function, who is calling this and how it is used?

Reply:
`init()` is not actually a local function; it has special semantics in Go. It is invoked automatically when a package is first loaded. See:  https://golangdocs.com/init-function-in-golang